packages/react/src/progress-indicator/ProgressIndicatorItem.tsx

```diff
@@ -1,140 +1,47 @@
-	 *
-	 * `started` = This step has been started and still needs to be completed.
-	 *
-	 * `todo` = This step is available for the user to start.
-	 */
-	status: ProgressIndicatorItemStatus;
-	/** Set this item as currently active. Only supporte for the `blocked`, `started`, and `doing` statuses */
-	isActive?: boolean;
-}
-
-export const ProgressIndicatorItemLink = ({
-	children,
-	isActive,
-	status,
-	...props
-}: ProgressIndicatorItemLinkProps) => (
-	<ProgressIndicatorItem
-		as={TextLink}
-		isActive={Boolean(isActive)}
-		status={status}
-		{...props}
-	>
-		{children}
-	</ProgressIndicatorItem>
-);
-
-export type ProgressIndicatorItemButtonProps =
-	ButtonHTMLAttributes<HTMLButtonElement> & {
-		background?: ProgressIndicatorBackground;
-		status: ProgressIndicatorItemStatus;
-		isActive: boolean;
-	};
-
-export const ProgressIndicatorItemButton = ({
-	children,
-	...props
-}: ProgressIndicatorItemButtonProps) => (
-	<ProgressIndicatorItem as={BaseButton} {...props}>
-		{children}
-	</ProgressIndicatorItem>
-);
-
-export const activeProgressStatusAllowedList: Array<ProgressIndicatorItemStatus> =
-	['blocked', 'doing', 'started'];
-
-export const activeProgressStatusBlockedList: Array<ProgressIndicatorItemStatus> =
-	['done', 'error', 'saved', 'todo'];
-export const allProgressStatuses: Array<ProgressIndicatorItemStatus> = [
-	...activeProgressStatusAllowedList,
-	...activeProgressStatusBlockedList,
-];
+	hoverColorMap,
+	ProgressIndicatorBackground,
+	progressIndicatorItemRingDataAttr,
+	ProgressIndicatorItemStatus,
+	progressIndicatorItemTextContainerDataAttr,
+	progressIndicatorItemTextDataAttr,
+	progressIndicatorItemTimelineDataAttr,
+	ProgressIndicatorItemWithLevelTwoItems,
+	statusMap,
+} from './utils';
 
 type ProgressIndicatorItemProps = PropsWithChildren<{
+	status: ProgressIndicatorItemStatus;
 	as: ElementType;
 	background?: ProgressIndicatorBackground;
-	status: ProgressIndicatorItemStatus;
-	isActive: boolean;
+	isActive?: boolean;
+	items?: ProgressIndicatorItemWithLevelTwoItems<{
+		isActive: boolean;
+	}>['items'];
 }>;
 
-const ProgressIndicatorItem = ({
+export const ProgressIndicatorItem = ({
 	as,
 	background = 'body',
 	children,
-	status,
 	isActive,
+	items,
+	status,
 	...props
 }: ProgressIndicatorItemProps) => {
-	const { label } = statusMap[status];
-
-	const listItemLinkTextSelector = '> span:last-of-type > span:first-of-type';
-
-	if (isActive && !activeProgressStatusAllowedList.includes(status)) {
-		throw new Error(
-			`The "${status}" status cannot be set to active. Please use one of the following status types instead: ${activeProgressStatusAllowedList.join(
-				', '
-			)}`
-		);
-	}
+	const { label: statusLabel } = statusMap[status];
 
 	useEffect(() => {
 		if (status === 'doing') {
 			console.warn(
-				'The "doing" status is deprecated. Use the "started" status with "isActive: true" applied instead.'
+				'The "doing" status is deprecated. Use the "started" status with `activePath`.'
```

Comment:
We should also warn(and deprecate) the use of the `isActive` key

Reply:
- [x] warn about deprecated `isActive` usage